### ISSUE TITLE
Why there are platform dependent configuration file selection?

### ISSUE THREAD

**Issue by tbhaxor** [Bash-it/bash-it#2084]
I see, there are platform dependent configurations for the bash-it entry point. For example in Darwin (macOS) it's `~/.bash_profile` and for other os it's `~/.bashrc` file.

https://github.com/Bash-it/bash-it/blob/4dbe92e38d1bfdbe8eba95af5e7d475621343742/install.sh#L189-L196

Point is why aren't we using only one file, that is ~/.bashrc? 

Based on the information provided [here](https://linuxize.com/post/bashrc-vs-bash-profile/) and dynamic module enable / disable feature that should allow bash to read and source the config on every terminal, I advice to use the rc file instead.

**Comment by gaelicWizard** [comment]
I agree 100% and have open PRs to address this, most directly #2039.

**Comment by tbhaxor** [comment]
@gaelicWizard I have added a comment on the PR

**Comment by seefood** [comment]
I believe I have covered this. closing.
